x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/aggregate/package-info.java

```diff
@@ -121,7 +121,7 @@
  *         It is also possible to declare any number of arbitrary arguments that must be provided via generated Supplier.
  *     </li>
  *     <li>
- *         {@code combine, combineStates, combineIntermediate, evaluateFinal} methods (see below) could be generated automatically
```

Comment:
I wonder if it would be worth it keeping the `combineStates()`, in case we want to reuse it later. For some node-level initial agg reduction. I think we talked about it at some point.
Not sure if it's really worth it. Commenting mostly because we're throwing quite some code.
However, I see there are no tests being removed though, so I suppose this wasn't tested either